meterfs/meterfs.c

```diff
@@ -843,90 +843,70 @@ static int meterfs_readRecord(file_t *f, void *buff, size_t bufflen, unsigned in
 {
 	/* This function assumes that f contains valid firstidx and firstoff */
 	unsigned int addr, pos, retry = 0;
-	unsigned char fbuff[32];
 	entry_t *eptr;
-	ssize_t rread, stat;
-	uint32_t checksum, echecksum;
-	void *checkbuff;
+	ssize_t stat, err = -ENOENT;
+	unsigned char *tmpptr;
 
-	if (f->firstidx.nvalid || idx > f->recordcnt)
+	/* Static buffer, realloced to fit the biggest record */
+	static unsigned char *rbuff = NULL;
+	static size_t rbuffsz = 0;
+
+	if (f->firstidx.nvalid || idx > f->recordcnt) {
 		return -ENOENT;
+	}
 
 	/* Calculate record position in a storage */
 	pos = (f->firstoff / (f->header.recordsz + sizeof(entry_t))) + idx;
 	pos = pos % ((f->header.sectorcnt * ctx->sectorsz) / (f->header.recordsz + sizeof(entry_t)));
 	addr = pos * (f->header.recordsz + sizeof(entry_t)) + f->header.sector * ctx->sectorsz;
 
-	if (bufflen > f->header.recordsz - offset)
+	if (bufflen > f->header.recordsz - offset) {
 		bufflen = f->header.recordsz - offset;
+	}
```

Comment:
`offset > f->header.recordsz` handling should be added here

Reply:
It's handled in readFile function (only one to call readRecord (https://github.com/phoenix-rtos/phoenix-rtos-filesystems/blob/3574e7bbd1fcfc6d7610c07b54a263237d498771/meterfs/meterfs.c#L1173)

Reply:
Yes it is.